polars/polars-time/src/chunkedarray/rolling_window/rolling_kernels/no_nulls.rs

```diff
@@ -42,11 +46,29 @@ pub(crate) fn rolling_min<T>(
     time: &[i64],
     closed_window: ClosedWindow,
     tu: TimeUnit,
+    tz: Option<&TimeZone>,
 ) -> ArrayRef
 where
     T: NativeType + PartialOrd + IsFloat + Bounded + NumCast + Mul<Output = T>,
 {
-    let offset_iter = groupby_values_iter(period, offset, time, closed_window, tu);
+    let offset_iter = match tz {
+        #[cfg(feature = "timezones")]
+        Some(tz) => match tz.parse::<Tz>() {
+            Ok(tz) => groupby_values_iter(period, offset, time, closed_window, tu, Some(tz)),
+            Err(_) => match parse_offset(tz) {
+                Ok(tz) => groupby_values_iter(period, offset, time, closed_window, tu, Some(tz)),
+                Err(_) => unreachable!(),
+            },
+        },
+        _ => groupby_values_iter(
+            period,
+            offset,
+            time,
+            closed_window,
+            tu,
+            NO_TIMEZONE.copied(),
+        ),
+    };
```

Comment:
I wanted to take this parsing one level up, but couldn't figure out how to satisfy the borrow checker in 

https://github.com/pola-rs/polars/blob/26305c1e6b91e90e387b371ef7f6b1a9a23ff10f/polars/polars-time/src/chunkedarray/rolling_window/mod.rs#L230-L240

Reply:
I don't think we can parse it a level higher as the function is generic. E.g. `impl TimeZoneTrait` has two implementations.

To parse this a level higher means that the function we accept in `rolling_agg` must be generic over the timezone type. This is possible, but I'd leave that to an exercise after this PR.